### PR TITLE
fix(addEventListener): when called from the global scope

### LINKED ISSUE #190: `addEventListener` in strict mode
When `addEventListener` is called in strict mode, `var target = isWebWorker() && !this ? self : this` would evaluate to `undefined` in `patchEventTargetMethods()` resulting in an error.

Notes:
- in non strict mode, `this` would be set to `window` and there would be no error,
- `window.addEventListener()` would work in both strict & non-strict mode.

/ref https://github.com/angular/angular/issues/4531

### PR DESCRIPTION
**Pull request by vicb**
fixes #190 

@IgorMinar / @btford could you please review.

I'll check within the context of Angular & report here

/ref https://github.com/angular/angular/issues/4531

### CODE REVIEW

**Comment by vicb** [comment]
Also see the discussion in https://github.com/angular/zone.js/pull/184

/cc @matthewjh

**Comment by vicb** [comment]
Status:
- There is one issue with this PR on Android,
- I can get my my Angular env to work (most probably related to the node/npm upgrades).

I'll try to solve those tomorrow.

File: dist/zone-microtask.js [hunk 1]
```diff
@@ -1072,11 +1075,15 @@ function patchEventTargetMethods(obj) {
     var eventType = eventName + (useCapturing ? '$capturing' : '$bubbling');
     if (handler[boundFnsKey] && handler[boundFnsKey][eventType]) {
       var _bound = handler[boundFnsKey];
-      
       arguments[1] = _bound[eventType];
       delete _bound[eventType];
     }
-    var target = isWebWorker() && !this ? self : this;
+
+    // - Inside a Web Worker, `this` is undefined, the context is `global` (= `self`)
```
Comment (IgorMinar) [inline dist/zone-microtask.js]: remove the reference to "self", it only confuses the reader

**Comment by IgorMinar** [comment]
otherwise lgtm

**Pull request merged**